gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergOverwritePartitionsStep.java

```diff
@@ -57,6 +58,7 @@
 @Slf4j
 public class IcebergOverwritePartitionsStep implements CommitStep {
```

Comment:
We should put a comment/include in the javadoc that schema updates must be enforced to be backwards compatible by the catalog in the partition overwrite step otherwise previous partitions may become unreadable.